<launch>
  
  <param name="/use_sim_time" value="true"/>


  <!-- To un-compress image topics -->
  <node name="republish1" type="republish" pkg="image_transport" output="screen" args="compressed in:=/gopro/image_raw raw out:=/gopro/image_raw" />


  <!-- marker size in cm -->    
	<arg name="marker_size" default="20.32" />
	<arg name="max_new_marker_error" default="0.02" />
	<arg name="max_track_error" default="0.05" />
	<arg name="marker_resolution" default="10"/>

	<arg name="cam_image_topic" default="/gopro/image_raw" />
	<arg name="cam_info_topic" default="/gopro/image_raw/camera_info" />
	<arg name="output_frame" default="gopro" />

	<node name="ar_track_alvar" pkg="ar_track_alvar" type="individualMarkersNoKinect" respawn="false" output="screen">
		<param name="marker_size"           type="double" value="$(arg marker_size)" />
		<param name="max_new_marker_error"  type="double" value="$(arg max_new_marker_error)" />
		<param name="max_track_error"       type="double" value="$(arg max_track_error)" />
		<param name="output_frame"          type="string" value="$(arg output_frame)" />
		<!-- <param name="marker_resolution"         value="$(arg marker_resolution)" /> -->
		<param name="max_frequency" value="30"/>

		<remap from="camera_image"  to="$(arg cam_image_topic)" />
		<remap from="camera_info"   to="$(arg cam_info_topic)" />
	</node>

	<node name="record_tag" pkg='utils_ros' type='record_ar_pose.py' required='true' output="screen">
		<param name="marker_topic" value="/ar_pose_marker" />
		<param name="output_file" value="/home/bjoshi/utils_ws/tags_cavern2.txt" />
	</node>

</launch>